### PR TITLE
bugfix: host-precomuted plan function for blackwell fmha

### LINKED ISSUE #1103: bug: blackwell fmha varlen encounter illegal memory access/wrong outputs for some
The current tile scheduler padded each request's query/output to equal length and failed to skip empty query blocks.

A minimal reproducing example:
```python
import math

import pytest
import torch

import flashinfer
import flashinfer.triton
from flashinfer.utils import is_sm100a_supported

def test_blackwell_cutlass_fmha(
    batch_size,
    qo_len,
    kv_len,
    num_qo_heads,
    num_kv_heads,
    head_dim_qk,
    head_dim_vo,
    sm_scale,
    causal,
    dtype,
):
    if qo_len > kv_len and causal:
        pytest.skip("qo_len > kv_len and causal is not supported")

    if not is_sm100a_supported(torch.device("cuda")):
        pytest.skip("SM100A is not supported on this device")
    torch.manual_seed(42)
    q = torch.randn(
        batch_size * qo_len, num_qo_heads, head_dim_qk, dtype=dtype, device="cuda"
    )

    qo_indptr = torch.tensor([   0, 1274, 2568, 3915, 5194, 6498, 7839, 8192], device="cuda", dtype=torch.int32)

    k = torch.randn(
        batch_size * kv_len, num_kv_heads, head_dim_qk, dtype=dtype, device="cuda"
    )
    v = torch.randn(
        batch_size * kv_len, num_kv_heads, head_dim_vo, dtype=dtype, device="cuda"
    )
    kv_indptr = qo_indptr

    wrapper = flashinfer.prefill.BatchPrefillWithRaggedKVCacheWrapper(
        torch.empty(128 * 1024 * 1024, device="cuda", dtype=torch.uint8),
        kv_layout="NHD",
        backend="cutlass",
    )
    wrapper.plan(
        qo_indptr,
        kv_indptr,
        num_qo_heads,
        num_kv_heads,
        head_dim_qk,
        head_dim_vo=head_dim_vo,
        causal=causal,
        sm_scale=sm_scale,
        q_data_type=dtype,
        kv_data_type=dtype,
    )
    o, lse = wrapper.run(q, k, v, return_lse=True)
    print(o)


if __name__ == "__main__":
    test_blackwell_cutlass_fmha(
        7,
        8192,
        8192,
        32,
        32,
        192,
        128,
        1,
        True,
        torch.bfloat16,
    )
```

Thanks @Qiaolin-Yu and @Fridge003 for contributing this example.

@yzh119 working on the new tile scheduler to fix the issue.

### PR DESCRIPTION
**Pull request by yzh119**
## 📌 Description

Accelerate blackwell causal fmha, and fix issue #1103

## 🔍 Related Issues

This PR should fix #1103 which is caused by variable sequence length, this PR skips the empty qo-tiles by precompute at host side.

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

### CODE REVIEW

File: flashinfer/prefill.py [hunk 1]
```diff
@@ -2495,6 +2494,21 @@ def run(
...
+            out, lse = fmha_varlen(
+                q=q,
+                k=k,
+                v=v,
+                qo_segment_offsets=self._qo_indptr_buf,
+                kv_segment_offsets=self._kv_indptr_buf,
+                plan_info=self._plan_info,
+                causal=self._causal,
+                sm_scale=sm_scale,
+                max_qo_len=max(self._qo_indptr_buf[1:] - self._qo_indptr_buf[:-1]),
```
Comment (yzh119) [inline flashinfer/prefill.py]: Please move it to `plan`, we don't want to synchronize for each `run`

**Review by cyx-6 — APPROVED** [review]

**Review by yongwww — APPROVED** [review]
